tests/visualization_tests/test_optimization_history.py

```diff
@@ -161,42 +185,30 @@ def objective(trial: Trial) -> float:
     studies = [create_study(direction=direction) for _ in range(n_studies)]
     for study in studies:
         study.optimize(objective, n_trials=3)
-    figure = plot_optimization_history(studies, error_bar=True)
-    assert len(figure.data) == 4
-    assert np.array_equal(figure.data[0].x, [0, 1, 2])
-    assert np.array_equal(figure.data[0].y, [1.0, 2.0, 0.0])
-    assert np.array_equal(figure.data[1].x, [0, 1, 2])
-    ydata = figure.data[1].y
-    if direction == "minimize":
-        assert np.array_equal(ydata, [1.0, 1.0, 0.0])
-    else:
-        assert np.array_equal(ydata, [1.0, 2.0, 2.0])
-    # Scatters for error bar don't have `name`.
-    legend_texts = [scatter.name for scatter in figure.data if scatter.name is not None]
-    assert sorted(legend_texts) == ["Best Value", "Objective Value"]
-    assert figure.layout.yaxis.title.text == "Objective Value"
```

Comment:
[Question] I couldn't find the test case corresponding to this one. Do we decide not to test it?
I think it is reasonable to remove it since we have a visual regression test now.

Reply:
Thanks for pointing it out! I think we should keep it as long as it is tied too much with visualization backends. I added tests of legends in `test_get_optimization_history_plot`.